url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/gui/ResultDisplay.java

```diff
@@ -0,0 +1,735 @@
+						}
+					}
+				}else if (tmp.equalsIgnoreCase("AMEND")){
+					updateLines++;
+					tmpPrice = set.getInt(6);
+					tmpVol = set.getInt(7);
+					if(tmpType.equalsIgnoreCase("B")){
+						tmpID = set.getLong(12);
+						myBidList.update(tmpID,tmpPrice,tmpVol,tmpTime);
+					}else if(tmpType.equalsIgnoreCase("A")){
+						tmpID = set.getLong(13);
+						myAskList.update(tmpID,tmpPrice,tmpVol,tmpTime);
+					}
+				}else if (tmp.equalsIgnoreCase("DELETE")){
+					deleteLines++;
+					if(tmpType.equalsIgnoreCase("B")){
+						tmpID = set.getLong(12);
+						myBidList.deleteOne(tmpID);
+					}else if(tmpType.equalsIgnoreCase("A")){
+						tmpID = set.getLong(13);
+						myAskList.deleteOne(tmpID);
+					}
+				}
+				count++;
+			};
+			for(int i = 0; i < strategyTrade.size();i++){
+				System.out.println("strategyTrade(" + i +") - Bid: " + strategyTrade.get(i).getBuyID() + " Ask: " 
+						+ strategyTrade.get(i).getAskID() + " Price: " + strategyTrade.get(i).getPrice());
+			}
+
+			LinkedList<resultObjectL> listOfResult;
+			System.out.println("listOfAllReciept size: " + strategy.getReceiptLength());
+			listOfResult = strategy.getResultListFromStrategy();
+
+			System.out.println("listOfResult size: " + listOfResult.size());
+
+			Mainmenu.console.append("Total lines read : " + count + "\n");
+			Mainmenu.console.append("Total lines matched : " + completedTrade.size() + " == tmpCount: " + tmpCount + "\n");
+			Mainmenu.console.append("Total lines update : " + updateLines + "\n");
+			Mainmenu.console.append("Total lines delete : " + deleteLines + "\n");
+			Mainmenu.console.append("Total Bid Error : " + myBidList.getError() + " Total Ask Error: " 
+					+ myAskList.getError()+ "\n");
+			Mainmenu.console.append("bid list contains " +  myBidList.getLength() + ".\n");
+			Mainmenu.console.append("ask list contains " +  myAskList.getLength() + ".\n");
+
+			resultObjectL tempResult;
+			double endResult = 0;
+			double endCount = 0;
+			while(!listOfResult.isEmpty()){
+				tempResult = listOfResult.poll();
+				if(tempResult.getPercentage() != Double.NEGATIVE_INFINITY){
+					endResult += tempResult.getPercentage();
+					endCount++;
+				}
+			}
+
+			//update jlabels
+			strategyResult.add(Integer.toString( count));
+			strategyResult.add(Integer.toString( completedTrade.size()));
+			strategyResult.add(Integer.toString( strategyCount));
+			strategyResult.add(Integer.toString( updateLines));
+			strategyResult.add(Integer.toString( deleteLines));
+			strategyResult.add(Integer.toString( myBidList.getLength()));
+			strategyResult.add(Integer.toString( myAskList.getLength()));
+			strategyResult.add(Double.toString(endResult/endCount));
+
+			set.close();
+		} catch (SQLException e) {
+			System.out.println("In Mainmenu/runStrategy : " + e);
+		}
+	}
+
+	private int matchTrade(MyBidList myBidList, MyAskList myAskList, LinkedList<ResultData> completedTrade, Time tmpTime, LinkedList<ResultData> strategyTrade) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.